lib/engine/game/g_1824.rb

```diff
@@ -127,13 +127,27 @@ def init_corporations(stock_market)
           end
         end
 
-        corporations.map do |corporation|
+        corporations.map! do |corporation|
           Engine::G1824::Corporation.new(
             min_price: stock_market.par_prices.map(&:price).min,
             capitalization: self.class::CAPITALIZATION,
             **corporation.merge(corporation_opts),
           )
         end
+
+        return corporations unless option_cisleithania
+
+        # Some corporations need to be removed, but they need to exists (for implementation reasons)
+        # So set them as closed and removed so that they do not appear
+        # Affected: Coal Railway C4 (SPB), Regional Railway BH and SB, and possibly UG
+        corporations.reject! do |c|
+          next unless %w[SPB SB BH].include?(c.id) || (two_player? && c.id == 'UG')
```

Comment:
don't call unless with multiple conditions

Reply:
Fixed. Was actually a bug as well so I rewrote this method.